odoo/addons/test_orm/models/test_schema.py

```diff
@@ -0,0 +1,78 @@
+from odoo import fields, models
+
+
+class TestSchema(models.Model):
+    _name = 'test_schema'
+    _description = 'Test Schema'
+
+    currency_id = fields.Many2one('res.currency')  # Needed for the monetary field.
+    res_model = fields.Char()  # Needed for the many2one_reference field.
+
+    # Binary Fields
+    binary_with_attachment = fields.Binary()
+    binary_without_attachment = fields.Binary(attachment=False)
+    image_with_attachment = fields.Image()
+    image_without_attachment = fields.Image(attachment=False)
+
+    # Misc Fields
+    boolean = fields.Boolean()
+    id = fields.Id()
+    json = fields.Json()
+
+    # Numeric Fields
+    integer = fields.Integer()
+    float_double_precision = fields.Float()
+    float_numeric = fields.Float(digits=(0, False))
+    monetary = fields.Monetary()
+
+    # Properties Fields
+    properties = fields.Properties()
+    properties_definition = fields.PropertiesDefinition()
+
+    # Reference Fields
+    reference = fields.Reference(selection=[('option_1', 'Option 1')])
+    many2one_reference = fields.Many2oneReference(model_field="res_model")
+
+    # Relational Fields
+    one2many = fields.One2many(comodel_name='test_schema.relations', inverse_name='many2one')
+    many2one = fields.Many2one(comodel_name="test_schema.relations")
+    many2many = fields.Many2many(comodel_name='test_schema.relations')
```

Comment:
I suggest to follow naming conventions: `many2one` → `many2one_id`, `one2many` → `one2many_ids`, etc.